### PR TITLE
Added new column  to sys.shards

### PR DESCRIPTION
**Pull request by matriv**
@dobe Already reviewed but please do the final review

### CODE REVIEW

File: blackbox/docs/sql/system.txt [hunk 1]
```diff
@@ -771,6 +771,16 @@ num_docs
...
 +----------------------+----------------------------------+-------------+
 
+min_lucene_version
+------------------
+
++------------------------+----------------------------------+-------------+
+|     Column Name        |           Description            | Return Type |
++========================+==================================+=============+
+| ``min_lucene_version`` | Shows the version of the oldest  | ``String``  |
+|                        | segment of the shard.            |             |
```
Comment (dobe) [inline blackbox/docs/sql/system.txt]: "Shows the oldest lucene segment version used in this shard"

background: this is not always the oldest segment

File: sql/src/main/java/io/crate/metadata/sys/SysShardsTableInfo.java [hunk 2]
```diff
@@ -160,6 +163,8 @@ public SysShardsTableInfo(ClusterService service, SysNodesTableInfo sysNodesTabl
                 .register(Columns.RECOVERY_FILES_PERCENT, FloatType.INSTANCE)
                 .register(Columns.PATH, DataTypes.STRING)
                 .register(Columns.BLOB_PATH, DataTypes.STRING)
+
```
Comment (dobe) [inline sql/src/main/java/io/crate/metadata/sys/SysShardsTableInfo.java]: newline :)

File: sql/src/main/java/io/crate/operation/reference/sys/shard/ShardMinLuceneVersionExpression.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    private IndexShard indexShard;
+
+    public ShardMinLuceneVersionExpression(IndexShard indexShard) {
+        this.indexShard = indexShard;
+    }
+
+    @Override
+    public BytesRef value() {
+        // This class is instanciated only once per shard so every query on sys.shards is calling value()
```
Comment (dobe) [inline sql/src/main/java/io/crate/operation/reference/sys/shard/ShardMinLuceneVersionExpression.java]: instantiated

File: sql/src/main/java/io/crate/operation/reference/sys/shard/ShardMinLuceneVersionExpression.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        // on the same object. We chose not to cache the value so in case you select min_lucene_version
+        // multiple times in the same statement like:
+        // `select min_lucene_version, min_lucene_version, ... from sys.shards`
+        // you might get different results.
+        try {
+            return new BytesRef(indexShard.minimumCompatibleVersion().toString());
```
Comment (dobe) [inline sql/src/main/java/io/crate/operation/reference/sys/shard/ShardMinLuceneVersionExpression.java]: does minimumCompatibleVersion  fit here?

File: blackbox/docs/sql/system.txt (hunk 1 above)
Comment (matriv) [inline blackbox/docs/sql/system.txt]: sorry don't get the background phrase...

File: sql/src/main/java/io/crate/metadata/sys/SysShardsTableInfo.java (hunk 2 above)
Comment (matriv) [inline sql/src/main/java/io/crate/metadata/sys/SysShardsTableInfo.java]: on purpose :-) but can remove

File: sql/src/main/java/io/crate/operation/reference/sys/shard/ShardMinLuceneVersionExpression.java (hunk 4 above)
Comment (matriv) [inline sql/src/main/java/io/crate/operation/reference/sys/shard/ShardMinLuceneVersionExpression.java]: Yep, we had a discussion with @seut too about this.

Comment (dobe) [inline sql/src/main/java/io/crate/operation/reference/sys/shard/ShardMinLuceneVersionExpression.java]: hm i am confused, you and me looked at the method it returns the lucene version of the index creation when no segment is there, which does not help.however if there are no segments, then it is probably no porble anyways, so ok to use it.

File: CHANGES.txt [hunk 5]
```diff
@@ -5,6 +5,9 @@ Changes for Crate
 Unreleased
 ==========
 
+ - Added new column ``min_lucene_version`` to ``sys.shards`` table,
+   which shows the version of the oldest segment of the shard.
```
Comment (dobe) [inline CHANGES.txt]: same here as in the docs regarding oldest

File: sql/src/main/java/io/crate/operation/reference/sys/shard/ShardMinLuceneVersionExpression.java (hunk 4 above)
Comment (matriv) [inline sql/src/main/java/io/crate/operation/reference/sys/shard/ShardMinLuceneVersionExpression.java]: @dobe yes this exactly what we discussed

**Pull request merged**